### PR TITLE
[sn-platform] feat: make sn-platform deployment idempotent

### PR DESCRIPTION
**Pull request by urfreespace**
*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: https://github.com/streamnative/charts/issues/556

### Motivation

Along with another PR https://github.com/streamnative/charts/pull/646/files, sn-platform deployment idempotent  is guaranteed 

Related issue: https://github.com/streamnative/charts/issues/556

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by fantapsody — APPROVED** [review]
LGTM. Does the init script ignore resources that already exist?

**Comment by urfreespace** [comment]
> LGTM. Does the init script ignore resources that already exist?

yes, in fact,  the initial API layer determines whether the resources already exists.

**Review by tuteng — APPROVED** [review]

File: charts/sn-platform/templates/streamnative-console/streamnative-console-initialize.yaml [hunk 1]
```diff
@@ -17,11 +17,16 @@
...
 
-{{- if .Values.initialize }}
 {{- if .Values.components.streamnative_console }}
 apiVersion: batch/v1
 kind: Job
 metadata:
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install,post-upgrade
```
Comment (MarvinCai) [inline charts/sn-platform/templates/streamnative-console/streamnative-console-initialize.yaml:27]: hmmmm, is `post-upgrade` a proper condition here? even we removed the .Values.initialize condition, the initialization job will only run once
if all the auth secrets are removed after a `helm upgrade` wouldn't pods fail to load secret when restarting?

**Review by MarvinCai — APPROVED** [review]

File: charts/sn-platform/templates/streamnative-console/streamnative-console-initialize.yaml [hunk 2]
```diff
@@ -17,11 +17,16 @@
...
+    "helm.sh/hook-weight": "5"
```
Comment (jdmaguire) [inline charts/sn-platform/templates/streamnative-console/streamnative-console-initialize.yaml:28]: A comment on why the hook weight is this value can be useful albeit not necessary. (ex. "# Run after FooBar")

**Review by jdmaguire — APPROVED** [review]

**Review by maxsxu — APPROVED** [review]

File: charts/sn-platform/templates/streamnative-console/streamnative-console-initialize.yaml (hunk 2 above)
Comment (urfreespace) [reply charts/sn-platform/templates/streamnative-console/streamnative-console-initialize.yaml:28]: thanks @jdmaguire , in fact, it's not necessary here, I will remove it.

**Pull request merged**